File: exercises/dotnet/McrDigital.Bootcamp1.Cards.Tests/AnimalDeckTest.cs
using FluentAssertions;

namespace McrDigital.Bootcamp1.Cards.Tests
{
    using Xunit;
    using McrDigital.Bootcamp1.Cards;

    public class AnimalDeckTest
    {
        [Fact]
        public void ShouldGetDeckInOrderAndReturn52CardsCorrectlyFormatted()
        {
            var deck = new AnimalDeck();
            var expectedCards = new string[]
            {
                "Aardvark",
                "Aardvark",
                "Baboon",
                "Baboon",
                "Camel",
                "Camel",
                "Deer",
                "Deer",
                "Elephant",
                "Elephant",
                "Frog",
                "Frog",
                "Gorilla",
                "Gorilla",
                "Hare",
                "Hare",
                "Impala",
                "Impala",
                "Jaguar",
                "Jaguar",
                "Kangaroo",
                "Kangaroo",
                "Lion",
                "Lion",
                "Moose",
                "Moose",
                "Newt",
                "Newt",
                "Octopus",
                "Octopus",
                "Penguin",
                "Penguin",
                "Quetzal",
                "Quetzal",
                "Rabbit",
                "Rabbit",
                "Salmon",
                "Salmon",
                "Tortoise",
                "Tortoise",
                "Uakaris",
                "Uakaris",
                "Vaquita",
                "Vaquita",
                "Whale",
                "Whale",
                "XRayTetra",
                "XRayTetra",
                "Yak",
                "Yak",
                "Zebra",
                "Zebra"
            };

            var actualCards = deck.GetCards();
            actualCards.Should().BeEquivalentTo(expectedCards, options => options.WithStrictOrdering());
        }

        [Fact]
        public void ShouldShuffleAndReturnCardsInAnyOrder()
        {
            var deck = new AnimalDeck();
            deck.Shuffle();

            var fullCardSetInAlphabeticalOrder = new string[]
            {
                "Aardvark",
                "Aardvark",
                "Baboon",
                "Baboon",
                "Camel",
                "Camel",
                "Deer",
                "Deer",
                "Elephant",
                "Elephant",
                "Frog",
                "Frog",
                "Gorilla",
                "Gorilla",
                "Hare",
                "Hare",
                "Impala",
                "Impala",
                "Jaguar",
                "Jaguar",
                "Kangaroo",
                "Kangaroo",
                "Lion",
                "Lion",
                "Moose",
                "Moose",
                "Newt",
                "Newt",
                "Octopus",
                "Octopus",
                "Penguin",
                "Penguin",
                "Quetzal",
                "Quetzal",
                "Rabbit",
                "Rabbit",
                "Salmon",
                "Salmon",
                "Tortoise",
                "Tortoise",
                "Uakaris",
                "Uakaris",
                "Vaquita",
                "Vaquita",
                "Whale",
                "Whale",
                "XRayTetra",
                "XRayTetra",
                "Yak",
                "Yak",
                "Zebra",
                "Zebra"
            };
            var actualCards = deck.GetCards();

            actualCards.Should().BeEquivalentTo(fullCardSetInAlphabeticalOrder).And.NotContainInOrder(fullCardSetInAlphabeticalOrder);
        }
    }
}

File: exercises/kotlin/src/test/kotlin/cards/AnimalDeckTest.kt
package cards

import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test
import kotlin.test.assertEquals
import kotlin.test.assertNotEquals

class AnimalDeckTest {

    @Test
    fun getDeckInOrder_return52CardsCorrectlyFormatted() {
        val animalDeck = AnimalDeck()
        val formattedList = listOf(
            "AARDVARK",
            "AARDVARK",
            "BABOON",
            "BABOON",
            "CAMEL",
            "CAMEL",
            "DEER",
            "DEER",
            "ELEPHANT",
            "ELEPHANT",
            "FROG",
            "FROG",
            "GORILLA",
            "GORILLA",
            "HARE",
            "HARE",
            "IMPALA",
            "IMPALA",
            "JAGUAR",
            "JAGUAR",
            "KANGAROO",
            "KANGAROO",
            "LION",
            "LION",
            "MOOSE",
            "MOOSE",
            "NEWT",
            "NEWT",
            "OCTOPUS",
            "OCTOPUS",
            "PENGUIN",
            "PENGUIN",
            "QUETZAL",
            "QUETZAL",
            "RABBIT",
            "RABBIT",
            "SALMON",
            "SALMON",
            "TORTOISE",
            "TORTOISE",
            "UAKARIS",
            "UAKARIS",
            "VAQUITA",
            "VAQUITA",
            "WHALE",
            "WHALE",
            "X_RAY_TETRA",
            "X_RAY_TETRA",
            "YAK",
            "YAK",
            "ZEBRA",
            "ZEBRA"
        )
        assertEquals(formattedList, animalDeck.getCards())
    }

    @Test
    fun shuffle_returnsAllCardsInAnyOrder() {
        val deck = AnimalDeck()
        deck.shuffle()
        val formattedList = listOf(
            "AARDVARK",
            "AARDVARK",
            "BABOON",
            "BABOON",
            "CAMEL",
            "CAMEL",
            "DEER",
            "DEER",
            "ELEPHANT",
            "ELEPHANT",
            "FROG",
            "FROG",
            "GORILLA",
            "GORILLA",
            "HARE",
            "HARE",
            "IMPALA",
            "IMPALA",
            "JAGUAR",
            "JAGUAR",
            "KANGAROO",
            "KANGAROO",
            "LION",
            "LION",
            "MOOSE",
            "MOOSE",
            "NEWT",
            "NEWT",
            "OCTOPUS",
            "OCTOPUS",
            "PENGUIN",
            "PENGUIN",
            "QUETZAL",
            "QUETZAL",
            "RABBIT",
            "RABBIT",
            "SALMON",
            "SALMON",
            "TORTOISE",
            "TORTOISE",
            "UAKARIS",
            "UAKARIS",
            "VAQUITA",
            "VAQUITA",
            "WHALE",
            "WHALE",
            "X_RAY_TETRA",
            "X_RAY_TETRA",
            "YAK",
            "YAK",
            "ZEBRA",
            "ZEBRA"
        )
        assertTrue(
            deck.getCards().containsAll(
                formattedList
            )
        )
        assertNotEquals(deck.getCards(), formattedList)
    }
}